### ISSUE TITLE
problem in running pygor command

### ISSUE THREAD

**Issue by decenwang** [penuts7644/TcrDataComparison#2]
Hi @penuts7644 
This is Decen. Sorry for trouble. However, I would like to get your help. I have installed the pygor module by the comman line `"pip install ./pygor"`, when I tried to parse the "bestscenarios.csv" by the "bestscenarios.py", I got nothing.
the code is :`python /path/to/bestscenarios.py ouput.csv.file`. it cannot move forward and give some errors. as the link (https://github.com/qmarcou/IGoR/issues/48) shows. Could you please inform me what's wrong with my code? if possible, please post the right codes. Thanks a lot!

Cheers,

Decen

**Comment by penuts7644** [comment]
He Decen,

No problem at all, I really understand you struggle with IGoR and pygor. For pygor, you need to know that all the code pieces are snippets (there is no command line tool). Meaning that you'll have to import them in your own code and use the correct functions inside the files.

For the `bestscenarios.py` file you'll need to use the following function located in the file: `read_bestscenarios_values`. Also, besides the bestscenarios file (generated by IGoR) you'll need a generative model object (by reading in the model parameters text file created by IGoR).

In Summary, build and IGoR model first (consists of model marginals file and parameters file). Use that model to calculate best scenarios. Make a python script that imports `pygor.counters.bestscenarios` and executes the `read_bestscenarios_values` function with the scenarios file and model parameters file.

Even though the [TcrDataComparison](https://github.com/penuts7644/TcrDataComparison) repo was just a small practice package, it does contain a [python file](https://github.com/penuts7644/TcrDataComparison/blob/master/python/IgorOutputTranslator.py) that executes the bestscenarios function from pygor **through command line**.

Cheers, Wout

**Comment by decenwang** [comment]
Hi Wout,


Sorry for delayed reply. Thanks a lot! Recently, I am busy on my report, after this, I will keep up IGoR. I expect your favor! Thanks!


Best,


Decen


| |
Decen (Qingkai Wang)
SUMC ＆ UNISS
|
|
+86 13651961997
wangqk198738@163.com
1152 East Huaxia Rd. Shanghai, China
For your best!
|
签名由网易邮箱大师定制
On 4/12/2019 12:25，Wout van Helvoirt<notifications@github.com> wrote：

He Decen,

No problem at all, I really understand you struggle with IGoR and pygor. For pygor, you need to know that all the code pieces are snippets (there is no command line tool). Meaning that you'll have to import them in your own code and use the correct functions inside the files.

For the bestscenarios.py file you'll need to use the following function located in the file: read_bestscenarios_values. Also, besides the bestscenarios file (generated by IGoR) you'll need a generative model object (by reading in the model parameters text file created by IGoR).

In Summary, build and IGoR model first (consists of model marginals file and parameters file). Use that model to calculate best scenarios. Make a python script that imports pygor.counters.bestscenarios and executes the read_bestscenarios_values function with the scenarios file and model parameters file.

Even though the TcrDataComparison repo was just a small practice package, it does contain a python file that executes the bestscenarios function from pygor through command line.

Cheers, Wout
